### ISSUE TITLE
Print only failing syscalls (the opposite of -z option)

### ISSUE THREAD

**Issue by esyr** [strace/strace#50]
From https://sourceforge.net/p/strace/feature-requests/3/

> An option to be able to check why a program which do not print any error message fail.
> On old version of strace (4.4.90) the option -z for "only succeeding syscalls". Well the opposite options would be a lot more helpfull :)

Depends on #49.

**Comment by esyr** [comment]
Maybe, more generic syscall return code filter is appropriate (or file it as a separate bug?).

**Comment by imz** [comment]
Just not to forget in case it gets implemented:

one case of a failure (which would explain the failure in Gtk Print dialog which I have been debugging) is `POLLNVAL` set by `poll`, e.g.:

    [pid  7330] poll([{fd=4<anon_inode:[eventfd]>, events=POLLIN}, {fd=5<socket:[84155]>, events=POLLIN}, {fd=3<socket:[84151]>, events=POLLIN}, {fd=18, events=POLLOUT}], 4, 0) = 1 ([{fd=18, revents=POLLNVAL}])

**Comment by esyr** [comment]
> `= 1`

this looks quite successful to me.

**Comment by imz** [comment]
Yes, it does formally look like a success, I agree.

But as for the stated purpose of the feature ("An option to be able to check why a program which do not print any error message fail."), it is another example of an effect of a syscall which means that something failed.

I don't insist on including this case into the suggested option/filter; this might be not formally correct. But it seems like a related wish; useful if looking for output which gives hints why a program failed.

          POLLERR
               Error condition (only returned in revents; ignored
               in events).

          POLLNVAL
               Invalid  request: fd  not open  (only returned  in
               revents; ignored in events).

Compare to a similar failure of `connect`:

     EBADF
          sockfd is not a valid open file descriptor.

**Comment by imz** [comment]
Actually, my own case of debugging followed exactly these levels of "filtering" the output of strace  (even without using this option): first, I looked through the socket operations in the strace output and looked for errors. I didn't see any errors (and the `poll` syscalls were formally a success, so I didn't pay attention) and couldn't understand what went wrong. After some time, I came to an idea what might be wrong (by other means, not strace), and only then I returned to the output of strace and noticed these error values; and this was the explanation indeed.
